### ISSUE TITLE
Using optional Self in method signatures fails

### ISSUE THREAD

**Issue by alexmacnorthstar** [astral-sh/ty#2042]
### Summary

```py
from typing import Self

class Foo:
	bar: Self | None

	def __init__(self, bar: Self | None):
		self.bar = bar

x = Foo(None)
```
Gives
```
error[invalid-assignment]: Object of type `Self@__init__ | None` is not assignable to attribute `bar` of type `<special-form 'typing.Self'> | None`
 --> test.py:7:3
  |
6 |     def __init__(self, bar: Self | None):
7 |         self.bar = bar
  |         ^^^^^^^^
  |
info: rule `invalid-assignment` is enabled by default
```

### Version

ty 0.0.2 (42835578d 2025-12-16)

**Comment by AlexWaygood** [comment]
Thanks!
